---
description: コンパイラ エラー CS0103
title: コンパイラ エラー CS0103
ms.date: 07/20/2015
f1_keywords:
- CS0103
helpviewer_keywords:
- CS0103
ms.assetid: fd1f2104-a945-4dba-8137-8ef869826062
ms.openlocfilehash: 689cebb258fcd75287e8e54630387ae7a6e986c5
ms.sourcegitcommit: d579fb5e4b46745fd0f1f8874c94c6469ce58604
ms.translationtype: HT
ms.contentlocale: ja-JP
ms.lasthandoff: 08/30/2020
ms.locfileid: "89134732"
---
# <a name="compiler-error-cs0103"></a>コンパイラ エラー CS0103

名前 'identifier' は現在のコンテキスト内に存在していません

 クラス、[名前空間](../keywords/namespace.md)、スコープに存在しない名前を使用しようとしました。 名前のスペルを確認し、使用中のディレクティブとアセンブリの参照を確認して、使用しようとしている名前が利用できることを確認します。

 このエラーは、ループあるいは `try` または `if` ブロックに変数を宣言し、外側のコード ブロックまたは別のコード ブロックから変数にアクセスしようとするときに頻繁に発生します。次の例をご覧ください。

```csharp
using System;

class MyClass1
{
    public static void Main()
    {
        try
        {
            // The following declaration is only available inside the try block.
            var conn = new MyClass1();
        }
        catch (Exception e)
        {  
            // The following expression causes error CS0103, because variable
            // conn only exists in the try block.
            if (conn != null)
                Console.WriteLine("{0}", e);
        }
    }
}
```

 次の例では、エラーを解決しています。

```csharp
using System;

class MyClass2
{
    public static void Main()
    {
        // To resolve the error in the example, the first step is to
        // move the declaration of conn out of the try block. The following
        // declaration is available throughout the Main method.
        MyClass2 conn = null;
        try
        {
            // Inside the try block, use the conn variable that you declared
            // previously.
            conn = new MyClass2();
        }
        catch (Exception e)
        {
            // The following expression no longer causes an error, because
            // the declaration of conn is in scope.
            if (conn != null)
                Console.WriteLine("{0}", e);
        }
    }
}
```
